vault_program/src/delegate_token_account.rs

```diff
@@ -0,0 +1,94 @@
+use jito_account_traits::AccountDeserialize;
+use jito_jsm_core::loader::{
+    load_signer, load_token_2022_program, load_token_account, load_token_mint, load_token_program,
+};
+use jito_vault_core::vault::Vault;
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, msg, program::invoke_signed,
+    program_error::ProgramError, pubkey::Pubkey,
+};
+
+/// Processes the delegate token account instruction: [`crate::VaultInstruction::DelegateTokenAccount`]
+///
+/// Admin might call the instruction when the vault is airdropped or transferred tokens
+pub fn process_delegate_token_account(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    amount: u64,
+) -> ProgramResult {
+    let [vault_info, admin, token_mint, token_account, delegate, token_program_info] = accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    Vault::load(program_id, vault_info, false)?;
+    load_signer(admin, false)?;
+    load_token_mint(token_mint)?;
+
+    let mut vault_data = vault_info.data.borrow_mut();
+    let vault = Vault::try_from_slice_unchecked_mut(&mut vault_data)?;
+
+    vault.check_admin(admin.key)?;
```

Comment:
nit: can you move this check below the account loading?

Reply:
Yeah!, Moved the code of checking admin after all accounts loaded.